### PR TITLE
Add Tagset + Profile modules; removed Tag module - closes Server-160

### PR DESCRIPTION
**Pull request by techsmyth**
Add Tagset module
Add Profile module
Remove Tag module

### CODE REVIEW

File: src/domain/agreement/agreement.entity.ts [hunk 1]
```diff
@@ -2,17 +2,19 @@ import { Field } from '@nestjs/graphql/dist/decorators/field.decorator';
...
 import { ID } from '@nestjs/graphql/dist/scalars';
 import { Project } from '../project/project.entity';
-import { Tag } from '../tag/tag.entity';
 import {
   BaseEntity,
   Column,
   Entity,
+  JoinColumn,
   JoinTable,
   ManyToMany,
```
Comment (valentinyanakiev) [inline src/domain/agreement/agreement.entity.ts:11]: Remove JoinTable, ManyToMany - not used

File: src/domain/agreement/agreement.entity.ts [hunk 2]
```diff
@@ -35,20 +37,15 @@ export class Agreement extends BaseEntity implements IAgreement {
...
+  @Field(() => Tagset, { nullable: true, description: 'The set of tags for the agreement' })
+  @OneToOne(() => Tagset, { eager: true, cascade: true })
+  @JoinColumn()
+  tagset: Tagset;
 
   constructor(name: string) {
     super();
     this.name = name;
+    this.tagset = new Tagset(RestrictedTagsetNames.Default);
+    //this.tagsetService.initialiseMembers(this.tagset);
```
Comment (valentinyanakiev) [inline src/domain/agreement/agreement.entity.ts:49]: Will this be used?

Comment (techsmyth) [reply src/domain/agreement/agreement.entity.ts:49]: will handle the whole initialisation path as a separate issue. Good to leave in for now as a reminder.

File: src/domain/organisation/organisation.entity.ts [hunk 3]
```diff
@@ -3,7 +3,6 @@ import { Challenge } from '../challenge/challenge.entity';
 import { DID } from '../did/did.entity';
 import { Ecoverse } from '../ecoverse/ecoverse.entity';
 import { IGroupable } from '../../interfaces/groupable.interface';
```
Comment (valentinyanakiev) [inline src/domain/organisation/organisation.entity.ts:5]: Remove JoinTable import - not used

File: src/domain/profile/profile.module.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Module } from '@nestjs/common';
+import { ProfileResolver } from './profile.resolver';
+
+@Module({
+  providers: [ProfileResolver],
```
Comment (valentinyanakiev) [inline src/domain/profile/profile.module.ts:5]: There is a provider in this module - I am assuming it needs to be exported.

File: src/domain/project/project.entity.ts [hunk 5]
```diff
@@ -1,18 +1,20 @@
 import { Field, ID, ObjectType } from '@nestjs/graphql';
 import { Agreement } from '../agreement/agreement.entity';
 import { Challenge } from '../challenge/challenge.entity';
-import { Tag } from '../tag/tag.entity';
 import {
   BaseEntity,
   Column,
   Entity,
+  JoinColumn,
   JoinTable,
```
Comment (valentinyanakiev) [inline src/domain/project/project.entity.ts:9]: JoinTable,
  ManyToMany,

--> They need to be removed

File: src/domain/tagset/tagset.service.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    return tagset;
+  }
+
+  async createRestrictedTagsets(tagsetable: ITagsetable, names: string[]): Promise<boolean> {
+    if (!tagsetable.restrictedTagsetNames) {
+      throw new Error('Non-initialised tagsetable submitted');
+    }
+    for (const name of names) {
+      const tagset = new Tagset(name);
```
Comment (valentinyanakiev) [inline src/domain/tagset/tagset.service.ts:24]: createRestrictedTagsets is a very odd method. You iterate over names, create  a new Tagset and never use it. I presume that's not the idea.

File: src/domain/user-group/user-group.interface.ts [hunk 7]
```diff
@@ -1,12 +1,11 @@
...
 export interface IUserGroup {
   id: number;
   name: string;
   members?: IUser[];
   focalPoint?: IUser;
-  tags?: ITag[];
-  //addUserToGroup(newUser: IUser): IUser;
-  //initialiseMembers(): IUserGroup;
+  profile?: IProfile;
+  //profile: IProfile;
```
Comment (valentinyanakiev) [inline src/domain/user-group/user-group.interface.ts:10]: Remove the second profile

File: src/domain/user/user.module.ts [hunk 8]
```diff
@@ -3,6 +3,7 @@ import { UserService } from './user.service';
 import { UserResolver } from './user.resolver';
 
 @Module({
+  imports: [],
```
Comment (valentinyanakiev) [inline src/domain/user/user.module.ts:6]: You don't need imports if you are not using it.

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Overall OK with some minor changes required. No issues design-wise.

File: src/domain/agreement/agreement.entity.ts (hunk 2 above)
Comment (valentinyanakiev) [reply src/domain/agreement/agreement.entity.ts:49]: I saw and recalled our conversation earlier - hence I haven't commented on the rest of the files 👍

**Pull request merged**